lib/framework/procbot.ts

```diff
@@ -172,32 +176,39 @@ export class ProcBot {
 	 */
 	protected dispatchToAllEmitters(data: ServiceEmitRequest): Promise<ServiceEmitResponse[]> {
 		let results: ServiceEmitResponse[] = [];
+		let emitterPromises: Array<Promise<void>> = [];
 
 		// If there's not a context for a particular emmiter, it will result in a response
 		// with an error contained with in specifying as such. It is up to clients to determine
 		// whether this is an issue or not
-		return Promise.map(this.emitters, (emitter) => {
-			return emitter.sendData(data)
-			.then((result) => { results.push(result); })
-			.catch((error) => { results.push(error); });
-		}).return(results);
+		this.emitters.forEach((emitterEntry) => {
+			const emitter = <ServiceEmitter>emitterEntry.instance;
+			const promise = emitter.sendData(data)
+				.then((result) => { results.push(result); })
+				.catch((error) => { results.push(error); });
+			emitterPromises.push(promise);
+		});
+
+		return Promise.all(emitterPromises).return(results);
```

Comment:
I think you can drop the `emitterPromises` and `results` arrays completely, and simplify this whole function to:

```typescript
return Promise.all(this.emitters.map((emitterEntry) => {
  const emitter = <ServiceEmitter>emitterEntry.instance // you can remove this too if we make ServiceMap generic
  return emitter.sendData(data).catch((err) => err)
}));
```